src/PhpStan/Type/Definition/Type/IsCompositeTypeStaticMethodTypeSpecifyingExtension.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\PhpStan\Type\Definition\Type;
+
+use GraphQL\Type\Definition\CompositeType;
+use GraphQL\Type\Definition\Type;
+use PhpParser\Node\Expr\StaticCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Analyser\SpecifiedTypes;
+use PHPStan\Analyser\TypeSpecifier;
+use PHPStan\Analyser\TypeSpecifierAwareExtension;
+use PHPStan\Analyser\TypeSpecifierContext;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StaticMethodTypeSpecifyingExtension;
+
+class IsCompositeTypeStaticMethodTypeSpecifyingExtension implements StaticMethodTypeSpecifyingExtension, TypeSpecifierAwareExtension
```

Comment:
```suggestion
final class IsCompositeTypeStaticMethodTypeSpecifyingExtension implements StaticMethodTypeSpecifyingExtension, TypeSpecifierAwareExtension
```

same for others

also, does this have to be in src? shouldn't we move it to tests?

Reply:
It doesn't really have anything to do with tests, and I think you probably want it to be distributed to consumers of the package so they can use and check the public method in their own code. That said, there may be a better way to handle this...

Reply:
php world has usualy a bit weird standard folder structure -> `src` and `tests`. Compared to java's apache standard it's a bit limiting

But yup, I though it was not supposed to be consumed 🤔 is it?

`final` should be there in any case.

Reply:
I did a run into a problem--these files were souring the coverage results. They don't execute at run time, so it doesn't really make sense for them to be factored in when computing coverage. For now I took your advice and moved them into test.

Next I did a little research--apparently what people do is create a dedicated package for their phpstan extensions, and if people want to use these extensions then they add that package to their `composer.json` and add it to the `includes` section of their `phpstan.neon`.

Reply:
Yea, it uses to be in another repository. Extension installer can be also easily leveraged then.

Reply:
Sounds good. Do you think we can host that here under the `webonyx` space?

Reply:
Another option would be to just get rid of the `Type::is...` family of methods. Most of them just wrap a vanilla call to `instanceof`, and the more elaborate ones just add a call to `getNamedType`, which is arguably obfuscation.